src/pages/OrdersPage.jsx

```diff
@@ -0,0 +1,44 @@
+import { Box } from '@mui/material';
+import { useEffect } from 'react';
+import { getAllMedications } from '../API/MedicationAPI';
+import { getAllOrders } from '../API/OrdersAPI';
+import OrdersTable from '../components/data-display/OrdersTable';
+import OrderForm from '../components/forms/OrderForm';
+import { useAppContext } from '../HOC/AppContext';
+
+const OrdersPage = () => {
+  const { ordersList, medicationsList, updateOrders, updateMedications } =
+    useAppContext();
+
+  useEffect(() => {
+    getAllOrders()
+      .then((orders) => {
+        updateOrders(orders.body);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+
+    getAllMedications()
+      .then((medications) => {
+        updateMedications(medications.body);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
```

Comment:
getAllOrders and getAllMedications won't throw exceptions, but if and error occurs during the fetching process for either one, they will return an http response object. 
```suggestion
  useEffect(() => {
    let ignoreOrders = false;
    let ignoreMedication = false;
    
    (async () => {
      const ordersResponse = await getAllOrders();
      if (response.ok && !ignoreOrders)
        updateOrders(ordersResponse.body);
      else
        //handle bad response
        
      const medicationResponse = await getAllMedications();
      if (medicationResponse.ok && !ignoreMedication)
        updateMedications(medicationResponse.body);
      else
        //handle bad response
    })();
    
    //cleanup
    return () => {
      ignoreOrders = true;
      ignoreMedication = true;
    }
  }, []);
```